pkg/logs/internal/launchers/journald/launcher.go

```diff
@@ -43,6 +45,8 @@ func (l *Launcher) Start(sourceProvider launchers.SourceProvider, pipelineProvid
 	l.sources = sourceProvider.GetAddedForType(config.JournaldType)
 	l.pipelineProvider = pipelineProvider
 	l.registry = registry
+	l.newJournalFn = func() (tailer.Journal, error) { return sdjournal.NewJournal() }
+	l.newJournalFromPathFn = func(path string) (tailer.Journal, error) { return sdjournal.NewJournalFromDir(path) }
```

Comment:
These closures do not capture anything. Would it be possible to change them to regular methods?

Reply:
These closure exist so we can inject a mock implementation of sdjournal. 

I tried assigning the method directly, but it seems the compiler doesn't like that
`l.newJournalFn = sdjournal.NewJournal`
```
cannot use sdjournal.NewJournal (value of type func() (j *sdjournal.Journal, err error)) as func() ("github.com/DataDog/datadog-agent/pkg/logs/internal/tailers/journald".Journal, error) value in assignment
```

However wrapping it in a closure seems to fix it despite `sdjournal.Journal` conforming to the interface.

Reply:
This arrangement means that the code in `Start` can not be tested (and the tests skip all initialization with a custom constructor). Would it make sense to separate this into a regular interface, and supply the implementation to the `NewLauncher`? Then the tests would be able to use regular API and test `NewLauncher` and `Start` as well.

Reply:
Note: I had to add 2 constructors: 
1. that includes the Journal Factory as an argument
2. that takes no arguments and constructs an `sdjournal` factory. 

Because in `agent.go` the constructor is called with no arguments and can either call the `systemd` version or the `no_systemd` version. [The interface has a dependency on the sdjounral library](https://github.com/DataDog/datadog-agent/pull/13874/files#diff-58411e2f20ca3ded4c76bf7590074321d0e0886fc631c85ad0995aa727847431R37) so I have to restrict this interface to code that includes the `systemd` build tag. Therefore it cannot be referenced in `agent.go`. 

I think this is still cleaner than it was before - but wanted to call this out for when you review it.